src/status_im/contexts/wallet/wallet_connect/events.cljs

```diff
@@ -90,15 +95,24 @@
 (rf/reg-event-fx
  :wallet-connect/on-session-request
  (fn [{:keys [db]} [event]]
-   (when (wallet-connect-core/event-should-be-handled? db event)
-     {:fx [[:dispatch [:wallet-connect/process-session-request event]]]})))
+   (if (wallet-connect-core/event-should-be-handled? db event)
+     {:fx [[:dispatch [:wallet-connect/process-session-request event]]]}
+     {:fx [[:dispatch
+            [:wallet-connect/send-response
+             {:error (wallet-connect/get-sdk-error
+                      constants/wallet-connect-user-rejected-chains-error-key)}]]]})))
 
 (rf/reg-event-fx
  :wallet-connect/on-session-delete
  (fn [{:keys [db]} [{:keys [topic] :as event}]]
-   (when (wallet-connect-core/event-should-be-handled? db event)
-     (log/info "Received Wallet Connect session delete: " event)
-     {:fx [[:dispatch [:wallet-connect/disconnect-session topic]]]})))
+   (if (wallet-connect-core/event-should-be-handled? db event)
+     (do
+       (log/info "Received Wallet Connect session delete: " event)
+       {:fx [[:dispatch [:wallet-connect/disconnect-session topic]]]})
+     {:fx [[:dispatch
+            [:wallet-connect/send-response
+             {:error (wallet-connect/get-sdk-error
+                      constants/wallet-connect-user-rejected-chains-error-key)}]]]})))
```

Comment:
Why do we need to send a response when we're getting notified by the app that session was deleted? We only need to delete it in database and that'd be it.

Reply:
yeah, we shouldn't be doing this

Reply:
This is a mistake, I'll remove it.